Add description on Tag form

mzagmajster opened the pull request:
<!-- ## Which branch should I use for my PR?

Assuming that:

a = current major release
b = current minor release
c = future major release

* a.x for any features and enhancements (e.g. 4.x)
* a.b for any bug fixes (e.g. 4.0, 4.1, 4.2)
* c.x for any features, enhancements or bug fixes with backward compatibility breaking changes (e.g. 5.x) -->

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [no]
| New feature/enhancement? (use the a.x branch)      | [yes]
| Deprecations?                          | [no]
| BC breaks? (use the c.x branch)        | [no]
| Automated tests included?              | [no] <!-- All PRs must maintain or improve code coverage -->
| Related user documentation PR URL      | mautic/mautic-documentation#... <!-- required for new features -->
| Related developer documentation PR URL | mautic/developer-documentation#... <!-- required for developer-facing changes -->
| Issue(s) addressed                     | Fixes #... <!-- prefix each issue number with "Fixes #", no need to create an issue if none exists, explain below instead -->

<!--
Additionally (see https://contribute.mautic.org/contributing-to-mautic/developer/code/pull-requests#work-on-your-pull-request):
 - Always add tests and ensure they pass.
 - Bug fixes must be submitted against the lowest maintained branch where they apply
   (lowest branches are regularly merged to upper ones so they get the fixes too.)
 - Features and deprecations must be submitted against the "4.x" branch.
-->

#### Description:

<!--
Please write a short README for your feature/bugfix. This will help people understand your PR and what it aims to do. If you are fixing a bug and if there is no linked issue already, please provide steps to reproduce the issue here.
-->

This PR adds summary and description on Tag form so the marketers can describe more precisely what given tag represents.

Note: We are now able to edit tag only the first time we open the form. If we want to rename the tag we have to create a new one, this however is not true for summary and description, we can update those two fields as much as we want (this is the expected behavior, you can read about reasoning behind this decision here: #10927).

I will try to add tests (and probably need some assistance on what scenario do I actually add) if codecov goes down.

#### Steps to test this PR:

<!--
This part is really important. If you want your PR to be merged, take the time to write very clear, annotated and step by step test instructions. Do not assume any previous knowledge - testers may not be developers.
-->
1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Go to new tag form
3. You can now add description to tag.

<!--
If you have any deprecations, list them here along with the new alternative.
If you have any backwards compatibility breaks, list them here.
-->


<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10799"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>

mzagmajster committed:
Add description to Mautic Tags

We wanted to be able to add summary and description of tags when we created them, so we extended tag
manager core plugin.

mzagmajster committed:
Add summary & desc. to Mautic Tags

We wanted to be able to add summary and description of tags when we created them, so we extended tag
manager core plugin.

mzagmajster committed:
Add TagManager migration file

mzagmajster committed:
Emable tag editing & cloning

Due to additional fields we had to adjust validation and form process a bit to allow summary and
description update after tag has been created.

mzagmajster committed:
Add type hints

mzagmajster committed:
Merge branch '4.x' into 242_extend_tags

kuzmany commented:
What is reason to make two fields for descriptions (description and summary)? From my point of view, description is enough

mzagmajster commented:
Summary is shown on list view and is short description. Description is
shown on details.

On Jan 28, 2022 13:07, "Zdeno Kuzmany" ***@***.***> wrote:

> What is reason to make two fields for descriptions (description and
> summary)? From my point of view, description is enough
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mautic/mautic/pull/10799#issuecomment-1024153083>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEKM5LW5G2K2QIFFGO7WR2DUYKBHZANCNFSM5MZUKVYQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

kuzmany commented:
As you see description is used in most entities (emails, segments, campaign, forms, stages...) and it's integrated without additional coding.. I vote to stay in that line for tags too and avoid differences between them.

mzagmajster commented:
Not sure what you mean by "integrated without additional coding", I had 
to add migration for description field so that the field can be added 
later. As far as summary field go, I can remove it if that would make it 
possible to accept this PR. I still need summary field to meet client 
requirements, but I guess I can add it via plugin functionality.

Tags originally did not have a description field.

Regards, M.

On 28/01/2022 17:22, Zdeno Kuzmany wrote:
>
> As you see description is used in most entities (emails, segments, 
> campaign, forms, stages...) and it's integrated without additional 
> coding.. I vote to stay in that line for tags too and avoid 
> differences between them.
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/mautic/mautic/pull/10799#issuecomment-1024375889>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AEKM5LRWDGTFU3QAPCDXNY3UYK7C7ANCNFSM5MZUKVYQ>.
> Triage notifications on the go with GitHub Mobile for iOS 
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> 
> or Android 
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>. 
>
> You are receiving this because you authored the thread.Message ID: 
> ***@***.***>
>

mzagmajster committed:
Update type hints

kuzmany commented:
I recommend stay in format what we've already use in each section.

Like in segments for example

![image](https://user-images.githubusercontent.com/462477/151665282-6c27fe55-0c8d-4b60-bdaa-2d39103e1930.png)

mzagmajster committed:
Remove summary field

So tags would be more consistent with the rest of the core components we removed the summary field
from Tag entity.

mzagmajster commented:
Here @kuzmany I have removed summary field from this contribution I hope its OK now, please check it and let me know. Thanks.

mzagmajster commented:
Not sure why this is failing now. Migration seems fine to me. Can someone provide an insight on why test is failing now?

RCheesley commented:
Hi @mzagmajster you can see the errors in the run details here: https://github.com/mautic/mautic/runs/5024024604?check_suite_focus=true#step:9:83

mzagmajster commented:
Hi, @RCheesley thank you, but these errors are not helpful to me. I added a simple migration to add description from the output of the tests its not clear to me how to fix it, it does not even seem to be related to my work.

Hmm... looking and thinking further does it complain the way it does because I fixed the migration I initially added a couple commits before on this branch and then with the latest commit I changed it? Maybe... I will check this out.

kuzmany commented:
@mzagmajster you need rebase your branch to actual 4.x
If it's more easy to you you can create new PR and cherry pick your commits

PR looks good and it's nice.  After clean up I will tests, hope we get it to 4.2. Thank you

kuzmany reviewed (dismissed):
Nice addition.
Works as describe.
We should add description to add/edit/view  functional tests workflow 

- https://github.com/mautic/mautic/blob/dcec76142986c4846cb2218bda6ea81d11a4f554/plugins/MauticTagManagerBundle/Tests/Functional/Controller/TagControllerTest.php#L155
- https://github.com/mautic/mautic/blob/dcec76142986c4846cb2218bda6ea81d11a4f554/plugins/MauticTagManagerBundle/Tests/Functional/Controller/TagControllerTest.php#L104
- https://github.com/mautic/mautic/blob/dcec76142986c4846cb2218bda6ea81d11a4f554/plugins/MauticTagManagerBundle/Tests/Functional/Controller/TagControllerTest.php#L82

mzagmajster committed:
Merge branch '242_extend_tags' of github.com:mzagmajster/mautic into 242_extend_tags

mzagmajster commented:
Hi, @kuzmany 

I tried to rebase this branch with the latest 4.x and check for mariadb is still failing with the same error. I suspect even if I create new PR this will still be the case. What should I do?

kuzmany commented:
@mzagmajster once If you resolve branch conflicts,  I am able to help you with finish it.

mzagmajster committed:
Merge branch '4.x' into 242_extend_tags

jos0405 committed:
Merge branch '4.x' into 242_extend_tags

jos0405 commented:
I tested this one, works as expected. I really like this addition.

jos0405 commented:
![image](https://user-images.githubusercontent.com/4246909/152531737-c3e3bcad-ebd3-44c5-82da-bebf8adc29a3.png)

jos0405 reviewed (dismissed):
Works as expected.
![image](https://user-images.githubusercontent.com/4246909/152532144-7598eff4-6222-4fc6-bc7d-4fd08b5ac6f0.png)

RCheesley committed:
Merge branch '4.x' into 242_extend_tags

kuzmany commented on app/bundles/LeadBundle/Entity/Tag.php:
```diff
@@ -56,6 +62,8 @@ public static function loadApiMetadata(ApiMetadataDriver $metadata)
                 [
                     'id',
                     'tag',
+                    'summary',
```
Summary does not exist

kuzmany commented on plugins/MauticTagManagerBundle/Controller/TagController.php:
```diff
@@ -332,9 +332,18 @@ private function createTagModifyResponse(Tag $tag, array $postActionVars, $actio
+                            break;
+                        }
+                    }
+                    //$found = $tagModel->getRepository()->countOccurrences($tag->getTag());
```
Do not need this line

mzagmajster replied:
I have resolved this.

kuzmany committed:
Merge branch '4.x' into 242_extend_tags

mzagmajster committed:
Cleanup after tags implementation

mzagmajster commented:
I have cleaned up as the comments suggest.

mzagmajster reviewed (commented):
I was able to add description text alongside tag name on tag creation.

bradycargle reviewed (approved):
Everything working for me, nice job!

kuzmany committed:
Merge branch '4.x' into 242_extend_tags

kuzmany committed:
Merge branch '4.x' into 242_extend_tags
